components/public-api/gitpod/experimental/v1/workspaces.proto

```diff
@@ -145,20 +145,31 @@ message WorkspaceStatus {
 message WorkspaceContext {
     // GitProvider describes the git provider
     message GitProvider {
-        // type is the git provider type, e.g. 'github', 'gitlab', 'bitbucked'
+        // type is the git provider type, e.g. 'github', 'gitlab', 'bitbucket'
         string type = 1;
 
         // hostname is the git provider hostname
         string hostname = 2;
     }
 
+    // Repository describes the originating repository
+    message Repository {
+        string name = 1;
+        string owner = 2;
+        string default_branch = 3;
+        string clone_url = 4;
+        bool private = 5;
+    }
+
     // Explicit Git context
     message Git {
         string normalized_context_url = 1;
-        string commit = 2;
+        string branch = 2;
```

Comment:
I don't think branch and commit should be here, instead we should have RepoStatus like in https://github.com/gitpod-io/gitpod/pull/18323/files#diff-cf04e8c717db8d475982fab101cffd340f3e6e79d689c04a9e996f82fdc7ecb3L298

Here we will only stable bits like repository. Or here it will be initial branch/commit but it does not make much sense, since we want to away from it.

cc @svenefftinge